pg_analytics/src/federation/executor.rs

```diff
@@ -0,0 +1,168 @@
+use crate::datafusion::context::DatafusionContext;
+use crate::datafusion::plan::create_logical_plan;
+use async_std::task;
+use async_trait::async_trait;
+use datafusion_federation_sql::SQLExecutor;
+use deltalake::datafusion::{
+    arrow::datatypes::{Schema, SchemaRef},
+    arrow::record_batch::RecordBatch,
+    error::{DataFusionError, Result},
+    physical_plan::{stream::RecordBatchStreamAdapter, SendableRecordBatchStream},
+};
+use pgrx::*;
+use std::sync::Arc;
+
+use crate::datafusion::datatype::{
+    DatafusionMapProducer, DatafusionTypeTranslator, PostgresTypeTranslator,
+};
+use crate::datafusion::table::DeltaTableProvider;
+use crate::errors::{NotFound, ParadeError};
+
+pub struct ColumnExecutor {}
+
+impl ColumnExecutor {
+    pub fn new() -> Result<Self> {
+        Ok(Self {})
+    }
+}
+#[async_trait]
```

Comment:
I think we're on Rust 1.75, we might not need `async_trait` anymore.

Reply:
It doesn’t compile when I get rid of it :(